### PR TITLE
push to stable

### PR DESCRIPTION
**Pull request by vanhauser-thc**

### CODE REVIEW

File: custom_mutators/autotokens/README [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# autotokens
```
Comment (domenukk) [inline custom_mutators/autotokens/README]: Capital A?

File: docs/custom_mutators.md [hunk 2]
```diff
@@ -132,13 +136,21 @@ def deinit():  # optional for Python
     for a specific queue entry, use this function. This function is most useful
     if `AFL_CUSTOM_MUTATOR_ONLY` is **not** used.
 
+- `splice_optout` (optional):
```
Comment (domenukk) [inline docs/custom_mutators.md:139]: How can this possibly work in python? Wouldn't the fn always be available?

File: docs/custom_mutators.md [hunk 3]
```diff
@@ -48,6 +48,7 @@ C/C++:
 ```c
 void *afl_custom_init(afl_state_t *afl, unsigned int seed);
 unsigned int afl_custom_fuzz_count(void *data, const unsigned char *buf, size_t buf_size);
+void afl_custom_splice_optout(void *data);
```
Comment (domenukk) [inline docs/custom_mutators.md:51]: Rust implementation?

File: src/afl-gotcpu.c [hunk 4]
```diff
@@ -214,7 +214,13 @@ int main(int argc, char **argv) {
   #if defined(__linux__)
       if (sched_setaffinity(0, sizeof(c), &c)) {
 
-        PFATAL("sched_setaffinity failed for cpu %d", i);
+        const char *error_code = "Unkown error code";
+        if (errno == EFAULT) error_code = "EFAULT";
```
Comment (domenukk) [inline src/afl-gotcpu.c:218]: switch case for legibility?

File: test/test-llvm.sh [hunk 5]
```diff
@@ -257,12 +257,13 @@ test -e ../afl-clang-fast -a -e ../split-switches-pass.so && {
   }
   rm -f test-compcov test.out instrumentlist.txt
   AFL_LLVM_CMPLOG=1 ../afl-clang-fast -o test-cmplog test-cmplog.c > /dev/null 2>&1
+  ../afl-clang-fast -o test-c test-cmplog.c > /dev/null 2>&1
```
Comment (domenukk) [inline test/test-llvm.sh:260]: Maybe add test-c to `.gitignore`?

**Review by domenukk — APPROVED** [review]
Not sure if the python custom mutator still works as intended, else looks good.
Rust bindings need support for the new custom mutator functions eventually

File: docs/custom_mutators.md (hunk 2 above)
Comment (vanhauser-thc) [reply docs/custom_mutators.md:139]: why would the function be available when it is not in the python script?  I dont know python really, but logically it does not make sense to me :)

File: docs/custom_mutators.md (hunk 3 above)
Comment (vanhauser-thc) [reply docs/custom_mutators.md:51]: uh can you do that? :) not sure what has to be done. pretty sure afl_custom_send is missing too

File: src/afl-gotcpu.c (hunk 4 above)
Comment (vanhauser-thc) [reply src/afl-gotcpu.c:218]: dunno what you mean :)

File: test/test-llvm.sh (hunk 5 above)
Comment (vanhauser-thc) [reply test/test-llvm.sh:260]: ok

File: src/afl-gotcpu.c (hunk 4 above)
Comment (domenukk) [reply src/afl-gotcpu.c:218]: Like
```c
switch (errno) {
    case EFAULT:
        error_code = "EFAULT";
        break;
    case ...
```

File: docs/custom_mutators.md (hunk 2 above)
Comment (domenukk) [reply docs/custom_mutators.md:139]: Do you check somewhere in the code if it's set in python?

File: src/afl-gotcpu.c (hunk 4 above)
Comment (vanhauser-thc) [reply src/afl-gotcpu.c:218]: was a PR, nobody is using the tool really ... yes that would be nicer, but I like to invest my time better :)

File: docs/custom_mutators.md (hunk 2 above)
Comment (vanhauser-thc) [reply docs/custom_mutators.md:139]: fixed

**Pull request merged**